CovertInfiltration/Src/CovertInfiltration/Classes/X2Helper_Infiltration_TemplateMod.uc

```diff
@@ -275,46 +281,63 @@ static function PatchTLPWeapons()
 
 static function PatchUtilityItems ()
 {
-	local X2ItemTemplateManager TemplateManager;
+	local X2ItemTemplateManager ItemTemplateManager;
+	local array<X2DataTemplate> DifficulityVariants;
+	local X2DataTemplate DataTemplate;
 	local X2ItemTemplate ItemTemplate;
-	local ArtifactCost Resources;
-
-	TemplateManager = class'X2ItemTemplateManager'.static.GetItemTemplateManager();
-	
-	ItemTemplate = TemplateManager.FindItemTemplate('Medikit');
-	if(ItemTemplate != none)
-	{
-		ItemTemplate.Cost.ResourceCosts.Length = 0;
-		Resources.ItemTemplateName = 'Supplies';
-		Resources.Quantity = default.COST_MEDIKIT_MK1;
-		ItemTemplate.Cost.ResourceCosts.AddItem(Resources);
-	}
-	
-	ItemTemplate = TemplateManager.FindItemTemplate('NanoMedikit');
-	if(ItemTemplate != none)
-	{
-		ItemTemplate.Cost.ResourceCosts.Length = 0;
-		Resources.ItemTemplateName = 'Supplies';
-		Resources.Quantity = default.COST_MEDIKIT_MK2;
-		ItemTemplate.Cost.ResourceCosts.AddItem(Resources);
-	}
-	
-	ItemTemplate = TemplateManager.FindItemTemplate('FlashbangGrenade');
-	if(ItemTemplate != none)
-	{
-		ItemTemplate.Cost.ResourceCosts.Length = 0;
-		Resources.ItemTemplateName = 'Supplies';
-		Resources.Quantity = default.COST_FLASHGRENADE;
-		ItemTemplate.Cost.ResourceCosts.AddItem(Resources);
-	}
+	local ItemCostOverride ItemCostOverride;
+	local name TemplateName;
+	local int TemplateDifficulty;
 	
-	ItemTemplate = TemplateManager.FindItemTemplate('SmokeGrenade');
-	if(ItemTemplate != none)
+	ItemTemplateManager = class'X2ItemTemplateManager'.static.GetItemTemplateManager();
+	`CI_Log("Overriding item costs");
+
+	foreach default.arrItemCostOverrides(ItemCostOverride)
 	{
-		ItemTemplate.Cost.ResourceCosts.Length = 0;
-		Resources.ItemTemplateName = 'Supplies';
-		Resources.Quantity = default.COST_SMOKEGRENADE;
-		ItemTemplate.Cost.ResourceCosts.AddItem(Resources);
+		DifficulityVariants.Length = 0;
+		ItemTemplateManager.FindDataTemplateAllDifficulties(ItemCostOverride.ItemName, DifficulityVariants);
+
+		foreach DifficulityVariants(DataTemplate)
+		{
+			ItemTemplate = X2ItemTemplate(DataTemplate);
+
+			if (ItemTemplate == none)
+			{
+				`CI_Warn(DataTemplate.Name @ "is not an X2ItemTemplate");
+				continue;
+			}
+
+			if (ItemTemplate.TemplateAvailability < 32)
+			{
+				continue; // Non-singleplayer template
+			}
+			else if (ItemTemplate.TemplateAvailability < 64)
```

Comment:
A better check would be `ItemTemplate.IsTemplateAvailableToAllAreas(BITFIELD_GAMEAREA_Rookie)`, `ItemTemplate.IsTemplateAvailableToAllAreas(BITFIELD_GAMEAREA_Veteran)`, etc. Hardcoding numbers like this is kinda very confusing